kv/kvnemesis: TestKVNemesisMultiNode failed

[(kv/kvnemesis).TestKVNemesisMultiNode failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2566064&tab=buildLog) on [master@d8b5cb0fdd5f0b1b28919539b78c17aee68dc5f4](https://github.com/cockroachdb/cockroach/commits/d8b5cb0fdd5f0b1b28919539b78c17aee68dc5f4):

```
          |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_command.go:763
          |   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).executeAdminBatch
          |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_send.go:575
          |   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).sendWithRangeID
          |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_send.go:100
          |   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).Send
          |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_send.go:36
          |   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).Send
          |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/store_send.go:195
          |   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Stores).Send
          |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/stores.go:177
          |   | github.com/cockroachdb/cockroach/pkg/server.(*Node).batchInternal.func1
          |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/node.go:874
          |   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
          |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:326
          |   | github.com/cockroachdb/cockroach/pkg/server.(*Node).batchInternal
          |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/node.go:862
          |   | github.com/cockroachdb/cockroach/pkg/server.(*Node).Batch
          |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/server/node.go:900
          |   | github.com/cockroachdb/cockroach/pkg/roachpb._Internal_Batch_Handler.func1
          |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/roachpb/api.pb.go:9083
          |   | github.com/cockroachdb/cockroach/pkg/util/tracing.ServerInterceptor.func1
          |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/tracing/grpc_interceptor.go:136
          |   | google.golang.org/grpc.getChainUnaryHandler.func1
          |   | 	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:921
          |   | github.com/cockroachdb/cockroach/pkg/rpc.NewServer.func1
          |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:176
          |   | google.golang.org/grpc.getChainUnaryHandler.func1
          |   | 	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:921
          |   | github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor
          |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/auth.go:60
          |   | google.golang.org/grpc.chainUnaryServerInterceptors.func1
          |   | 	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:907
          |   | github.com/cockroachdb/cockroach/pkg/roachpb._Internal_Batch_Handler
          |   | 	/go/src/github.com/cockroachdb/cockroach/pkg/roachpb/api.pb.go:9085
          |   | google.golang.org/grpc.(*Server).processUnaryRPC
          |   | 	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1082
          |   | google.golang.org/grpc.(*Server).handleStream
          |   | 	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:1405
          |   | google.golang.org/grpc.(*Server).serveStreams.func1.1
          |   | 	/go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:746
          |   | runtime.goexit
          |   | 	/usr/local/go/src/runtime/asm_amd64.s:1374
          | Wraps: (2) cannot merge ranges when lhs is in a joint state or has learners: (n1,s1):1,(n3,s3):2LEARNER
          | Error types: (1) *errbase.opaqueWrapper (2) *errutil.leafError
        Wraps: (7) merge failed: cannot merge ranges when lhs is in a joint state or has learners: (n1,s1):1,(n3,s3):2LEARNER
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *errbase.opaqueWrapper (4) *errutil.withPrefix (5) *errbase.opaqueWrapper (6) *secondary.withSecondaryError (7) *errutil.leafError
    kvnemesis_test.go:84: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode253320323
--- FAIL: TestKVNemesisMultiNode (1.42s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestKVNemesisMultiNode PKG=./pkg/kv/kvnemesis TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestKVNemesisMultiNode.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/kv/kvnemesis/kvnemesis_test.go
+++ b/pkg/kv/kvnemesis/kvnemesis_test.go
@@ -55,7 +55,6 @@ func TestKVNemesisSingleNode(t *testing.T) {
 
 func TestKVNemesisMultiNode(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	skip.WithIssue(t, 58624, "flaky test")
 	skip.UnderRace(t)
 
 	defer log.Scope(t).Close(t)

--- a/pkg/kv/kvnemesis/validator.go
+++ b/pkg/kv/kvnemesis/validator.go
@@ -310,7 +310,8 @@ func (v *validator) processOp(txnID *string, op Operation) {
 			// However, I think the right thing to do is sniff this inside the
 			// AdminMerge code and retry so the client never sees it. In the meantime,
 			// no-op. #44377
-		} else if resultIsError(t.Result, `merge failed: cannot merge range with non-voter replicas`) {
+		} else if resultIsError(t.Result,
+			`merge failed: cannot merge ranges when (rhs)|(lhs) is in a joint state or has learners`) {
 			// This operation executed concurrently with one that was changing
 			// replicas.
 		} else if resultIsError(t.Result, `merge failed: ranges not collocated`) {

